### PR TITLE
test: tests call `collate.SetNewCollationEnabledForTest` should run in serial

### LINKED ISSUE #29319: test: tests call  `collate.SetNewCollationEnabledForTest(true)` should serial
## Bug Report

we cannot make it parallel because of there is `collate.SetNewCollationEnabledForTest(true)` at L134

_Originally posted by @xiongjiwei in https://github.com/pingcap/tidb/pull/29125#discussion_r740047956_

### 1. Minimal reproduce step (Required)

see above

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #29319 

Problem Summary:

tests call `collate.SetNewCollationEnabledForTest` should run in serial.
checked all callers of whole repo and correct them by remove t.Parallel()

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/collation_test.go [hunk 1]
```diff
@@ -29,8 +29,6 @@ import (
 )
 
 func TestCompareString(t *testing.T) {
-	t.Parallel()
```
Comment (tisonkun) [inline expression/collation_test.go:32]: This is covered by #29276 . Please revert or rebase on #29276 .  I hope #29726 be merged soon.

File: planner/core/plan_to_pb_test.go [hunk 2]
```diff
@@ -27,7 +27,6 @@ import (
 )
 
 func TestColumnToProto(t *testing.T) {
-	t.Parallel()
```
Comment (tisonkun) [inline planner/core/plan_to_pb_test.go:30]: please move serial tests in `serial_test` file.

Comment (feitian124) [reply planner/core/plan_to_pb_test.go:30]: updated. please review again.

**Review by tisonkun — APPROVED** [review]
LGTM.

/cc @xiongjiwei

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**